src/database/DataTypes/Transaction.js

```diff
@@ -163,19 +163,17 @@ export class Transaction extends Realm.Object {
   }
 
   /**
-   * Delete any items that aren't contributing to this transaction, in order to
+   * Delete any batches and items that aren't contributing to this transaction, in order to
    * remove clutter
    * @param  {Realm} database   App wide local database
    * @return {none}
    */
-  pruneRedundantItems(database) {
-    const itemsToPrune = [];
-    this.items.forEach(transactionItem => {
-      if (transactionItem.totalQuantity === 0) {
-        itemsToPrune.push(transactionItem);
-      }
-    });
-    database.delete('TransactionItem', itemsToPrune);
+  pruneRedundantBatches(database) {
+    const batchesToRemove = this.getTransactionBatches(database)
+                              .filtered('numberOfPacks = 0');
+
+    database.delete('TransactionBatch', batchesToRemove);
```

Comment:
I found a bug, it wasn't evident when ItemBatches were deleted from within ExternalSupplierInvoices page.  As per issue #449, origin isExternalSupplierInvoice check in TransactionBatch destructor
Please change:
```
transaction.isExternalSupplierInvoice
return this.otherParty.isExternalSupplier
```
```
Name.isExternalSupplier
return this.type === 'facility' && this.isSupplier;
```